packages/rooks/src/hooks/useResizeObserverRef.ts

```diff
@@ -20,26 +19,35 @@ const config: ResizeObserverOptions = {
  * @see https://rooks.vercel.app/docs/useResizeObserverRef
  */
 function useResizeObserverRef(
-  callback: ResizeObserverCallback,
+  callback: ResizeObserverCallback | undefined,
   options: ResizeObserverOptions = config
 ): [CallbackRef] {
+  const { box } = options;
   const [node, setNode] = useState<HTMLElementOrNull>(null);
-  const freshCallback = useFreshTick(callback);
+  const callbackRef = useRef(callback);
+  callbackRef.current = callback;
```

Comment:
This should be in a useEffect since it is a side effect. 😄  It's not a big deal, but it makes sense semantically to keep side effects out of the render phase. 🙏
```suggestion
 useEffect(() => {
      callbackRef.current = callback;
 }) 
```

Reply:
You are right. In principle, side effects should not be in the render phase, but in useEffect.
Thanks!